src/redux/books/booksSlice.js

```diff
@@ -1,40 +1,120 @@
+  try {
+    const response = await axios.post(url, data, headers);
+    return response.data;
+  } catch (error) {
+    return error;
+  }
+});
+
+const fetchBooks = createAsyncThunk(
+  'books/fetchBooks',
+  async () => {
+    try {
+      const { data } = await axios.get(url);
+      return data;
+    } catch (err) {
+      return err.message;
+    }
+  },
+);
+
+const removeBook = createAsyncThunk('books/removeBook', async (id) => {
+  const data = JSON.stringify({ item_id: id });
+  try {
+    const response = await axios.delete(url + id, data, headers);
+    return response.data;
+  } catch (error) {
+    return error;
+  }
+});
 
 const booksSlice = createSlice({
   name: 'books',
-  initialState: {
-    books: [{
-      item_id: 'item1',
-      title: 'The Great Gatsby',
-      author: 'John Smith',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item2',
-      title: 'Anna Karenina',
-      author: 'Leo Tolstoy',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item3',
-      title: 'The Selfish Gene',
-      author: 'Richard Dawkins',
-      category: 'Nonfiction',
-    }],
-  },
-  reducers: {
-    appendBooks(state, action) {
-      state.books.push(action.payload);
-    },
-
-    removeBooks(state, action) {
-      return {
-        ...state,
-        books: state.books.filter((book) => book.item_id !== action.payload),
-      };
-    },
+  initialState,
+  reducers: {},
+  extraReducers: {
+    [addBooks.pending]: (state) => ({
+      ...state,
+      isLoading: true,
+    }),
+    [addBooks.fulfilled]: (state) => ({
+      ...state,
+      isLoading: false,
+      ifSucceed: false,
+    }),
+    [addBooks.rejected]: (state) => ({
+      ...state,
+      isLoading: false,
+    }),
+    [fetchBooks.pending]: (state) => ({
+      ...state,
+      isLoading: true,
+    }),
+    [fetchBooks.fulfilled]: (state, action) => ({
+      ...state,
+      isLoading: false,
+      ifSucceed: true,
+      books: action.payload,
+    }),
+    [fetchBooks.rejected]: (state) => ({
+      ...state,
+      isLoading: false,
+    }),
+    [removeBook.pending]: (state) => ({
+      ...state,
+      isLoading: true,
+    }),
+    [removeBook.fulfilled]: (state) => ({
+      ...state,
+      isLoading: false,
+      ifSucceed: false,
+    }),
+    [removeBook.rejected]: (state) => ({
+      ...state,
+      isLoading: false,
+    }),
   },
 });
 
-export const { appendBooks, removeBooks } = booksSlice.actions;
+// const booksSlice = createSlice({
+//   name: 'books',
+//   initialState: {
+//     books: [],
+//   },
+//   reducers: {
+//     appendBooks(state, action) {
+//       state.books.push(action.payload);
+//     },
+
+//     removeBooks(state, action) {
+//       return {
+//         ...state,
+//         books: state.books.filter((book) => book.item_id !== action.payload),
+//       };
+//     },
+//   },
+// });
+
+// export const { appendBooks, removeBooks } = booksSlice.actions;
```

Comment:
### **Optional**

- [ ] Please remove all commented code from your codebase to make your code clean and easy to maintain.